doc/contrib/python_packaging.rst

```diff
@@ -0,0 +1,53 @@
+#########################
+Notes on Python packaging
+#########################
+
+
+.. contents:: Contents
+  :local:
+
+.. _binary_wheels:
+
+***************************************************
+How to build binary wheels and source distributions
+***************************************************
+
+Binary wheels and source distributions (sdist for short) are two main
```

Comment:
I should remove the "binary" bit from this sentence here, since the sentence applies generally to all Python packages.

I'll also add a sentence somewhere saying that, for XGBoost specifically, `pip wheel` produces a binary wheel due to the existence of a compiled binary.